AnkiDroid/src/main/java/com/ichi2/anki/JavaScriptTTS.java

```diff
@@ -0,0 +1,103 @@
+package com.ichi2.anki;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.speech.tts.TextToSpeech;
+import java.util.Locale;
+
+// Since it is assumed that only advanced users will use the JavaScript api, 
+// here, Android's TextToSpeech is converted for JavaScript almost as it is, giving priority to free behavior.
+// https://developer.android.com/reference/android/speech/tts/TextToSpeech
+//
+// 
+
+public class JavaScriptTTS implements TextToSpeech.OnInitListener {
+
+    private TextToSpeech mTts;
+    private boolean mTtsOk;
+    private static final Bundle mTtsParams = new Bundle();
+
+    //The constructor will create a TextToSpeech instance.
+    JavaScriptTTS(Context context) {
+        mTts = new TextToSpeech(context, this);
+    }
+
+    @Override
+    //OnInitListener method to receive the TTS engine status
+    public void onInit(int status) {
+        if (status == TextToSpeech.SUCCESS) {
+            mTtsOk = true;
+        }
+        else {
+            mTtsOk = false;
+        }
```

Comment:
brackets are optional around the assignment
```suggestion
        mTtsOk = status == TextToSpeech.SUCCESS;
```